### PR TITLE
Support for spot instances on Azure

### PR DESCRIPTION
**Pull request by alexott**
This PR adds support for Azure spot instances for clusters & instance pools. It also adds an additional support for preemptible executors on GCP (only for clsuters)

### CODE REVIEW

File: compute/resource_cluster.go [hunk 1]
```diff
@@ -321,6 +324,12 @@ func modifyClusterRequest(clusterModel *Cluster) {
 		}
 		clusterModel.AwsAttributes = &awsAttributes
 	}
+	if clusterModel.AzureAttributes != nil {
+		clusterModel.AzureAttributes = nil
```
Comment (nfx) [inline compute/resource_cluster.go:328]: nit: can we add a unit test here

File: compute/resource_cluster.go [hunk 2]
```diff
@@ -321,6 +324,12 @@ func modifyClusterRequest(clusterModel *Cluster) {
...
+	}
+	if clusterModel.GcpAttributes != nil {
+		clusterModel.GcpAttributes = nil
```
Comment (nfx) [inline compute/resource_cluster.go:331]: nit: can we add a unit test here

File: compute/resource_instance_pool.go [hunk 3]
```diff
@@ -61,6 +61,8 @@ func ResourceInstancePool() *schema.Resource {
 		s["custom_tags"].ForceNew = true
 		s["enable_elastic_disk"].ForceNew = true
 		s["enable_elastic_disk"].Default = true
+		s["aws_attributes"].ConflictsWith = []string{"azure_attributes"}
+		s["azure_attributes"].ConflictsWith = []string{"aws_attributes"}
```
Comment (nfx) [inline compute/resource_instance_pool.go:65]: no gcp attributes for instance pools?

**Review by nfx — APPROVED** [review]

File: compute/resource_instance_pool.go (hunk 3 above)
Comment (alexott) [reply compute/resource_instance_pool.go:65]: there is no instance pools support yet, so I can't test it - will add later

File: compute/resource_cluster.go (hunk 2 above)
Comment (alexott) [reply compute/resource_cluster.go:331]: let me add it...

Comment (alexott) [reply compute/resource_cluster.go]: added

File: compute/resource_cluster.go (hunk 1 above)
Comment (nfx) [reply compute/resource_cluster.go:328]: But does it make sense to nil the field if it's not nil?..

Comment (alexott) [reply compute/resource_cluster.go:328]: all Azure attributes now are related to the Spot instances only, and if we're using instance pools, then we shouldn't override instance types?

**Review by nfx — APPROVED** [review]

**Pull request merged**